setup.py

```diff
@@ -188,14 +188,12 @@ def run(self):
         for package in mpl_packages:
             try:
                 result = package.check()
-                if result is not None:
-                    print_status(package.name, 'yes [%s]' % result)
-            except setupext.CheckFailed as e:
+            except setupext.Skipped as e:
                 print_status(package.name, 'no  [%s]' % str(e))
-                if not package.optional:
-                    sys.exit("Failed to build %s" % package.name)
-            else:
-                good_packages.append(package)
+                continue
+            if result is not None:
```

Comment:
`result` should always be a string according to `check()` docstring.

If you just want to make sure this gives nice output anyway an unconditional

```
print_status(package.name, f'yes [{result or ""}]')
```

seems better.

Otherwise, the meaning of None should be documented in `check()`.

Reply:
clarified the docstring (None means "print nothing" -- that's unchanged from previously).

Reply:
Oh, and there would be better names than result, e.g. message. 😀